packages/mock/src/client.ts

```diff
@@ -720,6 +720,17 @@ export class MockFetchClient {
       return exampleValueSet;
     }
 
+    // Special case for ServiceRequestTimeline and DefaultResourceTimeline Task query
+    // Since we don't support _filter yet
+    // TODO(ThatOneBro 24 Oct 2024): Remove this once we support _filter in the in-memory search implementation
+    if (
+      /^fhir\/R4\/Task\?_filter=based-on\+eq\+[a-zA-Z]+%2F[0-9a-zA-Z-]+\+or\+focus\+eq\+[a-zA-Z]+%2F[0-9a-zA-Z-]+\+or\+subject\+eq\+[a-zA-Z]+%2F[0-9a-zA-Z-]+/.test(
+        url
+      )
+    ) {
```

Comment:
Is there a less brittle way we might be able to handle this?

Reply:
Open to suggestions, these are hardcoded queries that show up everywhere in `examples` and `app` tests since they appear in both `ServiceRequestTimeline` and `DefaultResourceTimeline`

https://github.com/medplum/medplum/blob/main/packages/react/src/ServiceRequestTimeline/ServiceRequestTimeline.tsx#L23
https://github.com/medplum/medplum/blob/main/packages/react/src/DefaultResourceTimeline/DefaultResourceTimeline.tsx#L19

The idea is we don't want to return anything in too generic of a case so that it fails and triggers the error

Reply:
Yeah... I think this is might be too much of a special case.

We could silently ignore all `_filter`, but that seems like a weird middle ground.

I think the right answer is that we should implement `_filter` in `MemoryRepository` before we can ship this.